### DISCUSSION TITLE
v1.7.0 - AI, of course! But how?

### DISCUSSION THREAD

**Discussion by tehniemer** [mealie-recipes/mealie#3639] [Q&A] [answered]
Maybe I'm blind, but I can't find anywhere to add my OpenAI API key. I've hunted both in the docs and the application itself.

**Comment by Kuchenpirat** [comment, marked as answer]
You add your information through env vars: 

Here is the documentation for it: https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openai

**Comment by applesoff** [comment]
if i am selfhosting ollama, how do i get the API key that mealie needs?

**Reply by michael-genson** [reply]
[Ollama is specifically addressed in the docs](https://docs.mealie.io/documentation/getting-started/installation/open-ai/#configuration). The OpenAI client requires an API key, but Ollama does not actually authenticate with one, so you can pass any value and it should work
